src/com/amazon/corretto/crypto/provider/HpkeGen.java

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.*;
+import java.security.spec.AlgorithmParameterSpec;
+
+public class HpkeGen extends KeyPairGeneratorSpi {
+  private final AmazonCorrettoCryptoProvider provider_;
+  private HpkeParameterSpec spec = null;
+
+  HpkeGen(AmazonCorrettoCryptoProvider provider) {
+    Loader.checkNativeLibraryAvailability();
+    provider_ = provider;
+  }
+
+  /** Generates a new HPKE key and returns a pointer to it. */
+  private static native long generateEvpHpkeKemKeyFromSpec(int hpke_kem_id);
+
+  @Override
+  public void initialize(final AlgorithmParameterSpec params, final SecureRandom rnd)
+      throws InvalidAlgorithmParameterException {
+    if (params instanceof HpkeParameterSpec) {
+      // TODO: do validation
```

Comment:
I don't think this is necessary if you declare the HpkeParameterSpec constructor as private.

Reply:
Good catch.